source/client.js

```diff
@@ -411,17 +411,18 @@ export class ImmersClient extends window.EventTarget {
   }
 
   /**
-   * Delete  user's avatar from their Avatar Collection.
-   * @param  {(object|string)} sourceActivity - IRI of activity or Activity in the Outbox
+   * Delete a message.
+   * @param  {(string|Activities.APActivity)} sourceActivity - IRI of activity or Activity in the Outbox
    * @returns {Promise<string>} IRI of the remove activity
    */
   async deleteMessage (sourceActivity) {
-    switch (sourceActivity.type.toLowerCase()) {
+    const Activity = (typeof sourceActivity === 'string' ? this.activities.getObject(sourceActivity) : sourceActivity)
```

Comment:
@wmurphyrd this currently fails as object doesn't contain the Activity.type
we should either require APActivity, or create a new getActivity(fromIRI) method